### PR TITLE
Allowing Config class to resolve isFullscreen flag from CLI arguments

### PR DESCRIPTION
**Pull request by robyn-dressler**
Addresses feature request #1147

Changes:
- Move CLI argument parsing to `Config` class
- Added `-f` and `--fullscreen` flags that override `isFullscreen` in the config
- Added `patchFile` member variable to config (but excluding from TOML file, since it's game specific)
- Added second overload to `Emulator::Run`

I have unfortunately not been able to properly test fullscreen in my environment due to #1348, but I think the new CLI argument works.

### CODE REVIEW

File: src/emulator.cpp [hunk 1]
```diff
@@ -256,6 +256,14 @@ void Emulator::Run(const std::filesystem::path& file) {
     std::exit(0);
 }
 
+void Emulator::Run(int& argc, char* argv[]) {
```
Comment (SachinVin) [inline src/emulator.cpp]: Why is `argc` a reference?

**Comment by LNDF** [comment]
It might be better to use an arg parsing library like Boost program_options

**Comment by qurious-pixel** [comment]
linux-sdl appimage works when entering path `./Shadps4-sdl.AppImage /path/to/eboot.bin -f` or `--fullscreen`.
Used to `./Shadps4-sdl.AppImage --fullscreen /path/to/eboot.bin` order convention, but this fails with case-sensitive error message `Eboot.bin file not found`. Whereas removing the switch, it boots into the game.

File: src/emulator.cpp (hunk 1 above)
Comment (robyn-dressler) [reply src/emulator.cpp]: Good question, I don't know. I'll pass by value instead

**Comment by robyn-dressler** [comment]
@LNDF I really would like to use a library for this, but I was hesitant to introduce a new dependency to `common`, because of this comment in the style documentation:
> Don't ever introduce new external dependencies into Core

If that doesn't apply here, then I could rework this to use the Boost library. That would also probably resolve the ordering issue @qurious-pixel mentioned.

**Comment by LNDF** [comment]
Boost is already a dependency. You could try to open a PR in https://github.com/shadps4-emu/ext-boost adding boost/program_options.hpp using bcp.

**Comment by georgemoralis** [comment]
done in #1507  among other things

**Comment by DuckSleazzy** [comment]
I wasn't sure whether to open a new issue, or comment here.

On current release 0.5.0, running a game via CLI completely ignores `-f/--fullscreen`. I tried running Bloodborne with `shadps4.exe -g <path> -f true`. While the game itself boots fine, it is not fs. Later on pressing F11 makes it fs though.

Specs:
- Windows 11 24H2
- Ryzen 5800x
- RX 6650 XT
- 2x8GB 3600 CL18
- Aorus Gen4 1TB (OS), Samsung 860 EVO (Game and Emulator)

**Comment by ngoquang2708** [comment]
@DuckSleazzy It is fixed in the pre-release/nightly build.

**Pull request closed without merging**